packages/dom/CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Breaking Change
+
+- Deprecate `isNumberInput`, as it is no longer used internally.
```

Comment:
Nit: I usually include the link to PR if someone wants to explore the code:
```suggestion
- Deprecate `isNumberInput`, as it is no longer used internally ([#40896](https://github.com/WordPress/gutenberg/pull/40896)).
```

I found it useful many times in other OSS projects.